test/stdlib/TestCharacterSet.swift

```diff
@@ -289,8 +289,17 @@ class TestCharacterSet : TestCharacterSetSuper {
             expectNotEqual(abcSet, abcdSet)
             
             expectTrue(abcSet.isStrictSubset(of:abcdSet))
-            expectFalse(abcdSet.isStrictSubset(of:abcSet))
-            
+            #if (arch(i386) || arch(x86_64)) && os(iOS)
+            let vers = ProcessInfo.processInfo.operatingSystemVersion
+            if vers.majorVersion >= 10 && vers.minorVersion >= 3 && vers.patchVersion >= 0 {
+                expectFalse(abcdSet.isStrictSubset(of:abcSet))
+            } else {
+                print("running on \(vers) but was guarded to 10.3")
```

Comment:
it wouldn't just be those two items - it may be indicating that somehow the system version is being pulled in via linkage. The simulator has it's own sys root that should contain a version of CoreFoundation and Foundation that is the same as on device builds. If this is not the case we have a bigger problem than just CharacterSet. Hopefully it is just limited to the way the tests are built and run on the simulator for swift and not a more wide spread issue like the version of Foundation and CoreFoundation being completely ignore and the system version is somehow slipping in.